contracts/misc/src/bridge/LockboxUpgradeable.sol

```diff
@@ -0,0 +1,145 @@
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /**
+     * @dev Deposit tokens into the lockbox.
+     * @param value The amount of tokens to deposit.
+     */
+    function deposit(uint256 value) external whenNotPaused {
+        _deposit(msg.sender, msg.sender, value);
+    }
+
+    /**
+     * @dev Deposit tokens into the lockbox for a specific address.
+     * @param to    The address to deposit the tokens for.
+     * @param value The amount of tokens to deposit.
+     */
+    function depositTo(address to, uint256 value) external whenNotPaused {
+        _deposit(msg.sender, to, value);
+    }
+
+    /**
+     * @dev Withdraws tokens from the lockbox.
+     * @param value The amount of tokens to withdraw.
+     */
+    function withdraw(uint256 value) external whenNotPaused {
+        _withdraw(msg.sender, msg.sender, value);
+    }
+
+    /**
+     * @dev Withdraws tokens from the lockbox to a specific address.
+     * @param to    The address to withdraw the tokens to.
+     * @param value The amount of tokens to withdraw.
+     */
+    function withdrawTo(address to, uint256 value) external whenNotPaused {
+        _withdraw(msg.sender, to, value);
+    }
+
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                      ADMIN FUNCTIONS                       */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /**
+     * @dev Pauses depositing into and withdrawing from the lockbox.
+     */
+    function pause() external onlyRole(PAUSER_ROLE) {
+        _pause();
+    }
+
+    /**
+     * @dev Unpauses depositing into and withdrawing from the lockbox.
+     */
+    function unpause() external onlyRole(PAUSER_ROLE) {
+        _unpause();
+    }
+
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                     INTERNAL FUNCTIONS                     */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /**
+     * @dev Internal function to deposit tokens into the lockbox.
+     * @param from  The address of the depositor.
+     * @param to    The address to deposit the tokens for.
+     * @param value The amount of tokens to deposit.
+     */
+    function _deposit(address from, address to, uint256 value) internal {
+        token.safeTransferFrom(from, address(this), value);
+
+        IMintBurn(wrapper).mint(to, value);
+    }
+
+    /**
+     * @dev Internal function to withdraw tokens from the lockbox.
+     * @param from  The address of the withdrawer.
+     * @param to    The address to withdraw the tokens to.
+     * @param value The amount of tokens to withdraw.
+     */
+    function _withdraw(address from, address to, uint256 value) internal {
+        address _wrapper = wrapper; // Cache address for gas optimization.
+
+        _wrapper.safeTransferFrom(from, address(this), value);
+
+        IMintBurn(_wrapper).burn(value);
+
+        token.safeTransfer(to, value);
+    }
+}
```

Comment:
Should we be using rlusds `clawback(from, amount)` and `CLAWBACK_ROLE` to simplify burning? It would reduce number of approvals users need to do.

If yes, we can handle in separate PR.

It is a less standard interface. Tho I think it's okay to build with it in mind now. In future we could handle both `MintBurn` and `MintClawback` interfaces by using a little util

```solidity
function _burn(address from, uint256 amount) {
     if (clawback) {
         IMintBurn(token).clawback(from, amount)
     } else {
         token.safeTransferFrom(from, address(this), value);
         IMintBurn(token).burn(value);
     }
}
```